Student Sketches and Answers.jl

```diff
@@ -27,238 +26,320 @@ begin
 
-# ╔═╡ ca9c92f6-cb21-4b7a-b6b5-1268ffd97e3b
-# {{{This answer is not rendering correctly. Like with student 6, the \r\n gets displayed instead of starting a new line"
-Markdown.parse("""
-**Answer to Question 1**
+# ╔═╡ c46b5066-8128-4dc1-a71a-45973261aa82
+md"""**Final plot after answer 2**
 
-$text_answer_110 """)
+"""
 
-# ╔═╡ dea7b21e-f7ab-4fdf-8f80-099483f4cf2d
-begin
-	# {{{ Add a comment explaining what the function does. Name it something more informative than PLOT. Also, I think don't give a default to sy here. }}}
-	function PLOT(animation_data, sy=zeros(maximum(animation_data.xcor)-minimum(animation_data.xcor)+1))
-		x = animation_data.xcor
-		y = animation_data.ycor
-		xmin = 0
-		xmax = 20
-		ymin = -10
-		ymax = 10
-		sx=collect(xmin:xmax)
-
-	# {{{ I reformatted a how I want you to format multi-line function calls so they are easier to read }}}
-	p=plot(sx, sy,
-			lims=(xmin, xmax), 
-			ylims=(ymin, ymax),     
-			sseriestype = :line,  # Set the seriestype to line
+# ╔═╡ 065df6ae-57e6-4eaf-9397-1ba5d37b4acc
+md"""
+## Functions and data processing
+"""
+
+# ╔═╡ 4f1000dd-0e52-40bc-8d75-b2015e3e6198
+function build_sy_dataset(a_df, s_df)
+    sy_data_1 = DataFrame(uid = Int64[], timestamp = DateTime[], y_vector = Vector{Float64}[])
+    sy_data_2 = DataFrame(uid = Int64[], timestamp = DateTime[], y_vector = Vector{Float64}[])
+
+    function update_sy_vector(student_answers, qid, sketch_data, sy_data, student, current_sketch_y_vector = zeros(22))
+        if(size(student_answers, 1) == 2 || student_answers.q_id[1] == 110)
+            end_timestamp_1 = student_answers[student_answers.q_id .== 110, :].timestamp[1]
+        else
+            end_timestamp_1 = student_answers[student_answers.q_id .== 111, :].timestamp[1]
+        end
+
+        if qid == 110
+            animation_data = filter(row -> row.timestamp <= end_timestamp_1, sketch_data)
+            current_sketch_y_vector = zeros(22)
+        else
+            if(size(student_answers, 1) == 2 || student_answers.q_id[1] == 111)
+                end_timestamp_2 = student_answers[student_answers.q_id .== 111, :].timestamp[1]
+            else
+                end_timestamp_2 = sketch_data.timestamp[length(sketch_data.timestamp)]
+            end
+            animation_data = filter(row -> (end_timestamp_1 <= row.timestamp) && (row.timestamp <= end_timestamp_2), sketch_data)
+        end
+
+        x = animation_data.xcor
+        y = animation_data.ycor
+        t = animation_data.timestamp
+
+        for i in 1:size(animation_data, 1)
+            current_sketch_y_vector[x[i] + 1] = y[i]
+            push!(sy_data.uid, student)
+            push!(sy_data.timestamp, t[i])
+            push!(sy_data.y_vector, copy(current_sketch_y_vector))
+        end
+    end
+
+    for i in 1:length(unique(a_df.uid))
+        if unique(a_df.uid)[i] == 73 || unique(a_df.uid)[i] == 87 || unique(a_df.uid)[i] == 106 || unique(a_df.uid)[i] == 100 || unique(a_df.uid)[i] == 104
```

Comment:
I would rather not be manually excluding uids, because I want this code to work with future data as well. What are the criteria by which these are being excluded? Lets then exclude based on those criteria.